examples/gno.land/p/demo/context/context_test.gno

```diff
@@ -0,0 +1,96 @@
+package context
+
+import "testing"
+
+func TestContextExample(t *testing.T) {
+	type favContextKey string
+
+	k := favContextKey("language")
+	ctx := WithValue(Empty(), k, "Gno")
+
+	if v := ctx.Value(k); v != nil {
+		if string(v) != "Gno" {
+			t.Errorf("language value should be Gno, but is %s", v)
+		}
+	} else {
+		t.Errorf("language key value was not found")
+	}
+
+	if v := ctx.Value(favContextKey("color")); v != nil {
+		t.Errorf("color key was found")
+	}
+}
+
+// otherContext is a Context that's not one of the types defined in context.go.
+// This lets us test code paths that differ based on the underlying type of the
+// Context.
+type otherContext struct {
+	Context
+}
+
+type (
+	key1 int
+	key2 int
+)
+
+// func (k key2) String() string { return fmt.Sprintf("%[1]T(%[1]d)", k) }
+
+var (
```

Comment:
I'm not a huge fan of defining package-level vars for tests. Can we possibly move these in the tests themselves?

Reply:
For this specific case, I suggest letting things as they are. This part was ported directly from the original Go implementation: https://github.com/golang/go/blob/master/src/context/x_test.go#L206-L208